geoapps/inversion/components/factories/survey_factory.py

```diff
@@ -397,23 +402,59 @@ def _tdem_arguments(self, data=None, local_index=None, mesh=None):
 
         return [tx_list]
 
+    def _fem_arguments(self, data=None, mesh=None, channel=None):
+        frequencies = data.entity.channels if channel is None else [channel]
+        rx_locs = data.entity.vertices
+        tx_locs = data.entity.transmitters.vertices
+        freqs = data.entity.transmitters.workspace.get_entity("Tx frequency")[0]
+        freqs = np.array([int(freqs.value_map[f]) for f in freqs.values])
+
+        self.ordering = []
+        sources = []
+        rx_factory = ReceiversFactory(self.params)
+        tx_factory = SourcesFactory(self.params)
+        for receiver_id in self.local_index:
```

Comment:
Would make more sense to loop over the transmitters, in case you have multi receivers for the same tx. But we can change it when we tackle ground loop...

Reply:
done